spec/system/post_show_spec.rb

```diff
@@ -0,0 +1,73 @@
+RSpec.describe 'Post show page', type: :feature do
+  before(:each) do
+    @user = User.create(
+      name: 'berna',
+      bio: 'Aspiring FullStack Dev',
+      photo: 'https://via.placeholder.com/150',
+      posts_counter: 2
+    )
+
+    @user2 = User.create(
+      name: 'nati',
+      bio: 'FullStack Dev',
+      photo: 'https://via.placeholder.com/150',
+      posts_counter: 2
+    )
+
+    @post1 = Post.create(
+      title: 'Test',
+      text: 'First Post',
+      comments_counter: 0,
+      likes_counter: 0,
+      author: @user
+    )
+
+    @comment1 = Comment.create(
+      text: 'First Comment',
+      post: @post1,
+      author: @user2
+    )
+
+    @like1 = Like.create(
+      post: @post1,
+      author: @user2
+    )
+  end
+
+  describe 'the post show page' do
+    it 'displays the post title' do
+      visit user_post_path(@user, @post1)
+      expect(page).to have_content('Test')
+    end
+
+    it 'displays the post author' do
+      visit user_post_path(@user, @post1)
+      expect(page).to have_content('berna')
+    end
+
+    it 'displays number of comments' do
+      visit user_post_path(@user, @post1)
+      expect(page).to have_content('1')
+    end
+
+    it 'displays number of likes' do
+      visit user_post_path(@user, @post1)
+      expect(page).to have_content('1')
+    end
+
+    it 'displays the post text' do
+      visit user_post_path(@user, @post1)
+      expect(page).to have_content('First Post')
+    end
+
+    it 'displays the commentor' do
+      visit user_post_path(@user, @post1)
+      expect(page).to have_content('nati')
+    end
+
+    it 'displays the comment text' do
+      visit user_post_path(@user, @post1)
+      expect(page).to have_content('First Comment')
+    end
+  end
+end
```

Comment:
- [ ] Seven of the failures tests come from the `post_show_spec.rb`  please 🙏🏽 ensure proper creation order: Create the Comment records after the Post and User records are created. This order is necessary to ensure that the required associations and foreign keys are available during creation.

```bash
11) Post show page the post show page displays the post title
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_show_spec.rb:27:in `block (2 levels) in <top (required)>'

  12) Post show page the post show page displays the post author
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_show_spec.rb:27:in `block (2 levels) in <top (required)>'

  13) Post show page the post show page displays number of comments
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_show_spec.rb:27:in `block (2 levels) in <top (required)>'

  14) Post show page the post show page displays number of likes
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_show_spec.rb:27:in `block (2 levels) in <top (required)>'

  15) Post show page the post show page displays the post text
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_show_spec.rb:27:in `block (2 levels) in <top (required)>'

  16) Post show page the post show page displays the commentor
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_show_spec.rb:27:in `block (2 levels) in <top (required)>'

  17) Post show page the post show page displays the comment text
      Failure/Error:
        @comment1 = Comment.create(
          text: 'First Comment',
          post: @post1,
          author: @user2
        )
      
      ActiveModel::MissingAttributeError:
        can't write unknown attribute `author_id`
      # ./spec/system/post_show_spec.rb:27:in `block (2 levels) in <top (required)>'
```